wordpress-mobile/WordPress-iOS
# UL&S: Add `TextLinkTableViewCell` to SiteAddressViewController

Ref. #14009 
Ref. PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/308

This PR adds more base UI to the unified login & signup project. The `TextLinkTableViewCell` is used any time there is a hint button styled like a text link, such as "Reset your password" or "Find your site address". 

### To test:
1. `rake dependencies`
2. build and run
3. Log In > Or log in by entering your site address.
4. Observe the keyboard behavior: the site url textfield becomes first responder and the Continue button slides up.
5. Tap the "Find site address" text link button
6. Expected: the fancy alert message is presented.

### Screenshots
<a href="https://user-images.githubusercontent.com/1062444/85445859-705bc280-b559-11ea-928b-e9945a7f22a8.png"><img src="https://user-images.githubusercontent.com/1062444/85445859-705bc280-b559-11ea-928b-e9945a7f22a8.png" width="350" /></a>

<a href="https://user-images.githubusercontent.com/1062444/85445907-7baeee00-b559-11ea-9097-990e72a9ddaf.png"><img src="https://user-images.githubusercontent.com/1062444/85445907-7baeee00-b559-11ea-9097-990e72a9ddaf.png" width="350" /></a>

### Do not merge until:
1. Feature flag is disabled for `.unifiedSiteAddress`
2. The Authenticator PR has been merged: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/308
3. A new Authenticator release has been created
4. The release is published on Cocoapods trunk
5. This PR points to the new release

PR submission checklist:
